front_end/handlers/EditExerciseHandler.py

```diff
@@ -134,6 +135,12 @@ def post(self, course, assignment, exercise):
                                     if len(empty_tests) > 0:
                                         result = f"Warning: {len(empty_tests)} of your tests produced no output."
 
+                                    num_submissions = sum(list(map(lambda x: int(x[1]["num_submissions"]), self.content.get_exercise_scores(course, assignment, exercise))))
+
+                                    # If number of tests is greater than before last save and number of submissions on this exercise is greater than zero, raise warning.
+                                    if num_old_tests < len(tests) and num_submissions > 0:
+                                        result = f"Warning: You have increased the number of tests of an exercise that already has {num_submissions} submissions. This will render the output of those submissions unviewable for students. Their submission scores will not change."
```

Comment:
Minor thing, but please change the wording to "You have increased the number of tests, and this exercise already has {num_submissions} submissions. This will render the output of those submissions unviewable for students. However, their submission scores will not change."